experimental/traffic-portal/src/app/api/user.service.ts

```diff
@@ -267,6 +268,37 @@ export class UserService extends APIService {
 		return this.get<Array<ResponseRole>>(path).toPromise();
 	}
 
+	/**
+	 * Creates a new Role.
+	 *
+	 * @param role The role to create.
+	 * @returns The created role.
+	 */
+	public async createRole(role: RequestRole): Promise<ResponseRole> {
+		return this.post<ResponseRole>("roles", role).toPromise();
+	}
+
+	/**
+	 * Updates an existing Role.
+	 *
+	 * @param role The role to update.
+	 * @returns The updated role.
+	 */
+	public async updateRole(role: ResponseRole): Promise<ResponseRole> {
```

Comment:
This, unfortunately isn't enough information to update a Role. If you change the Role's name, this method won't work because it relies on the name of the Role _after_ editing - which must necessarily not exist. Therefore, the signature must be of the form

```typescript
public async updateRole(name: string, role: Role): Promise<PutResponseRole>
```
(`Role` is fine to use without `Response` on the front - in general that's going to be safe for any object that doesn't expose an ID).